test/TestParser.java

```diff
@@ -0,0 +1,134 @@
+import org.junit.rules.*;
+import org.junit.Rule;
+import org.junit.Ignore;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class TestParser {
+   private final String fileName = "test.txt";
+   private final String unicodeString = "Юникодтекст";
+   private final String nonUnicodeString = "Simple text";
+   @Rule
+   public TemporaryFolder folder = new TemporaryFolder();
+   public File saveStringToFile(String str) throws IOException {
+      File f = this.folder.newFile(this.fileName);
+      try {
+         PrintWriter out = new PrintWriter(f.getAbsoluteFile());
+         try {
+            out.print(str);
+         } finally {
+            out.close();
+         }
+      } catch(IOException e) {
+         throw new RuntimeException(e);
```

Comment:
Method already throws IOException, what is the purpose of re-throwing RuntimeException?